# Основы Python для обработки данных

Практическое занятие для начинающих аналитиков данных.

**Продолжительность:** 90 минут  
**Среда:** Google Colab, Jupyter Notebook или VS Code с поддержкой `.ipynb`

## Результат занятия

После занятия вы сможете:

- выполнять код в notebook;
- создавать переменные и определять типы данных;
- выполнять расчёты показателей;
- работать со строками, списками и словарями;
- использовать сравнения и логические выражения;
- применять `if`, `for` и `while`;
- решать небольшие аналитические задачи на данных о продажах.

## План занятия

| Блок | Время |
|---|---:|
| Работа в notebook, `print()`, комментарии | 10 мин |
| Переменные, типы данных и преобразование типов | 15 мин |
| Арифметика и расчёты аналитика | 15 мин |
| Строки и форматированный вывод | 10 мин |
| Сравнения и условия | 15 мин |
| Списки, словари и циклы | 20 мин |
| Итоговая мини-практика | 5 мин |

## 1. Как работать в notebook

Notebook состоит из ячеек двух типов:

- **Markdown** — текст, заголовки, пояснения и задания;
- **Code** — программный код.

Чтобы выполнить ячейку:

- нажмите кнопку запуска слева от ячейки;
- или нажмите **Shift + Enter**.

Важно выполнять ячейки сверху вниз. Иначе Python может не знать переменную, созданную в предыдущей ячейке.

In [1]:
print("Среда готова к работе")

Среда готова к работе


## 2. Функция `print()` и комментарии

`print()` выводит текст или значение на экран.

In [2]:
print("Здравствуйте!")
print("Начинаем изучать Python для аналитики данных.")

Здравствуйте!
Начинаем изучать Python для аналитики данных.


Комментарий начинается со знака `#`. Python не выполняет комментарии.

In [3]:
# Эта строка объясняет код
print("Эта строка выполняется")

Эта строка выполняется


### Пример 1. Вывод нескольких показателей

In [4]:
print("Количество заказов:", 12)
print("Выручка:", 54000)
print("Город:", "Казань")

Количество заказов: 12
Выручка: 54000
Город: Казань


## 3. Переменные

Переменная — это имя, под которым Python хранит значение.

В аналитике переменные могут хранить:

- цену;
- количество;
- выручку;
- название города;
- признак выполнения плана.

In [5]:
product_name = "Ноутбук"
unit_price = 72000
quantity = 3

print(product_name)
print(unit_price)
print(quantity)

Ноутбук
72000
3


### Правила хороших имён

Используйте:

- латинские буквы;
- понятные слова;
- нижнее подчёркивание между словами.

Хорошо:

```python
total_revenue = 120000
orders_count = 25
```

Плохо:

```python
a = 120000
x1 = 25
```

## 4. Основные типы данных

На первом этапе нужны четыре типа:

- `int` — целое число;
- `float` — число с дробной частью;
- `str` — строка;
- `bool` — логическое значение `True` или `False`.

In [6]:
orders_count = 12
average_check = 4250.75
city = "Казань"
plan_completed = True

print(type(orders_count))
print(type(average_check))
print(type(city))
print(type(plan_completed))

<class 'int'>
<class 'float'>
<class 'str'>
<class 'bool'>


### Пример 2. Типы данных в одной задаче

In [7]:
product = "Клавиатура"
price = 3500.0
quantity = 4
in_stock = True

print(f"Товар: {product}")
print(f"Цена: {price}")
print(f"Количество: {quantity}")
print(f"Есть на складе: {in_stock}")

Товар: Клавиатура
Цена: 3500.0
Количество: 4
Есть на складе: True


## 5. Преобразование типов

Иногда число загружается как текст. Тогда его нужно преобразовать.

Основные функции:

- `int()` — преобразовать в целое число;
- `float()` — преобразовать в дробное число;
- `str()` — преобразовать в строку.

In [8]:
quantity_text = "5"
price_text = "1250.50"

quantity = int(quantity_text)
price = float(price_text)

revenue = quantity * price

print(revenue)
print(type(quantity))
print(type(price))

6252.5
<class 'int'>
<class 'float'>


### Разбор типичной ошибки

Строки складываются как текст:

In [9]:
value_1 = "10"
value_2 = "20"

print(value_1 + value_2)

1020


Чтобы получить число `30`, строки нужно преобразовать:

In [10]:
result = int(value_1) + int(value_2)
print(result)

30


## 6. Арифметические операции

| Операция | Символ |
|---|---|
| Сложение | `+` |
| Вычитание | `-` |
| Умножение | `*` |
| Деление | `/` |
| Целая часть от деления | `//` |
| Остаток от деления | `%` |
| Возведение в степень | `**` |

In [11]:
print(10 + 5)
print(10 - 5)
print(10 * 5)
print(10 / 5)
print(11 // 5)
print(11 % 5)
print(2 ** 3)

15
5
50
2.0
2
1
8


### Пример 3. Выручка

In [12]:
unit_price = 2500
quantity = 3

revenue = unit_price * quantity

print("Выручка:", revenue)

Выручка: 7500


### Пример 4. Скидка

In [13]:
amount_before_discount = 15000
discount_rate = 0.10

discount_amount = amount_before_discount * discount_rate
final_amount = amount_before_discount - discount_amount

print("Сумма до скидки:", amount_before_discount)
print("Размер скидки:", discount_amount)
print("Итоговая сумма:", final_amount)

Сумма до скидки: 15000
Размер скидки: 1500.0
Итоговая сумма: 13500.0


### Пример 5. Средний чек

In [14]:
total_revenue = 84000
orders_count = 14

average_check = total_revenue / orders_count

print("Средний чек:", average_check)

Средний чек: 6000.0


### Округление

`round(value, 2)` округляет число до двух знаков после запятой.

In [15]:
total_revenue = 100000
orders_count = 17

average_check = total_revenue / orders_count

print(average_check)
print(round(average_check, 2))

5882.35294117647
5882.35


### Практика 1. Рассчитайте маржу

Дано:

- цена продажи — 5000 рублей;
- себестоимость — 3200 рублей;
- продано 8 единиц.

Найдите:

1. выручку;
2. общую себестоимость;
3. маржу.

In [16]:
# Сначала попробуйте решить самостоятельно

### Решение

In [17]:
unit_price = 5000
unit_cost = 3200
quantity = 8

revenue = unit_price * quantity
total_cost = unit_cost * quantity
margin = revenue - total_cost

print("Выручка:", revenue)
print("Себестоимость:", total_cost)
print("Маржа:", margin)

Выручка: 40000
Себестоимость: 25600
Маржа: 14400


## 7. Строки

Строка — это текст в кавычках.

In [18]:
city = "Москва"
category = "Электроника"

print(city)
print(category)

Москва
Электроника


Полезные методы строк:

- `.lower()` — нижний регистр;
- `.upper()` — верхний регистр;
- `.title()` — первая буква каждого слова заглавная;
- `.strip()` — удалить пробелы по краям;
- `.replace()` — заменить часть строки.

In [19]:
raw_city = "  мОСКВА  "

clean_city = raw_city.strip().title()

print(raw_city)
print(clean_city)

  мОСКВА  
Москва


In [20]:
category = "бытовая техника"
new_category = category.replace("бытовая", "мелкая бытовая")

print(new_category)

мелкая бытовая техника


### f-строки

f-строка соединяет текст и значения переменных.

In [21]:
city = "Самара"
orders_count = 18
revenue = 96500.5

print(f"Город: {city}")
print(f"Количество заказов: {orders_count}")
print(f"Выручка: {revenue:.2f} руб.")

Город: Самара
Количество заказов: 18
Выручка: 96500.50 руб.


### Практика 2. Сформируйте отчётную строку

Создайте переменные:

- `city = "Казань"`;
- `orders_count = 23`;
- `revenue = 127500.75`.

Выведите одну строку:

`Казань: 23 заказа, выручка 127500.75 руб.`

In [22]:
# Сначала попробуйте решить самостоятельно

### Решение

In [23]:
city = "Казань"
orders_count = 23
revenue = 127500.75

print(f"{city}: {orders_count} заказа, выручка {revenue:.2f} руб.")

Казань: 23 заказа, выручка 127500.75 руб.


## 8. Сравнения

Операции сравнения возвращают `True` или `False`.

- `>` — больше;
- `<` — меньше;
- `>=` — больше или равно;
- `<=` — меньше или равно;
- `==` — равно;
- `!=` — не равно.

In [24]:
revenue = 120000
plan = 100000

print(revenue > plan)
print(revenue >= plan)
print(revenue == plan)
print(revenue != plan)

True
True
False
True


## 9. Логические операторы

- `and` — оба условия должны быть истинными;
- `or` — достаточно одного истинного условия;
- `not` — меняет `True` на `False` и наоборот.

In [25]:
revenue = 120000
orders_count = 30

print(revenue >= 100000 and orders_count >= 25)
print(revenue >= 150000 or orders_count >= 25)
print(not revenue < 100000)

True
True
True


## 10. Условия `if`, `elif`, `else`

Условие позволяет выполнить разные действия в зависимости от результата проверки.

In [26]:
revenue = 120000
plan = 100000

if revenue >= plan:
    print("План выполнен")
else:
    print("План не выполнен")

План выполнен


В Python код внутри условия записывается с отступом в четыре пробела.

### Пример 6. Категория заказа

In [27]:
order_amount = 7800

if order_amount < 3000:
    category = "Небольшой заказ"
elif order_amount <= 10000:
    category = "Средний заказ"
else:
    category = "Крупный заказ"

print(category)

Средний заказ


### Пример 7. Два условия одновременно

In [28]:
revenue = 125000
returns_rate = 0.03

if revenue >= 100000 and returns_rate <= 0.05:
    status = "Хороший результат"
else:
    status = "Нужен дополнительный анализ"

print(status)

Хороший результат


### Практика 3. Оценка среднего чека

Правила:

- меньше 3000 — `Низкий`;
- от 3000 до 7000 включительно — `Средний`;
- больше 7000 — `Высокий`.

Используйте `average_check = 6400`.

In [29]:
# Сначала попробуйте решить самостоятельно

### Решение

In [30]:
average_check = 6400

if average_check < 3000:
    check_category = "Низкий"
elif average_check <= 7000:
    check_category = "Средний"
else:
    check_category = "Высокий"

print(check_category)

Средний


## 11. Списки

Список хранит несколько значений в определённом порядке.

In [31]:
daily_revenue = [12500, 9800, 14300, 11700, 15600]

print(daily_revenue)
print(type(daily_revenue))

[12500, 9800, 14300, 11700, 15600]
<class 'list'>


### Индексация и срезы

Индексация начинается с нуля.

In [32]:
print("Первое значение:", daily_revenue[0])
print("Второе значение:", daily_revenue[1])
print("Последнее значение:", daily_revenue[-1])
print("Первые три значения:", daily_revenue[:3])

Первое значение: 12500
Второе значение: 9800
Последнее значение: 15600
Первые три значения: [12500, 9800, 14300]


### Изменение списка

In [33]:
cities = ["Москва", "Казань"]

cities.append("Самара")
print(cities)

cities.remove("Казань")
print(cities)

['Москва', 'Казань', 'Самара']
['Москва', 'Самара']


### Полезные функции

In [34]:
daily_revenue = [12500, 9800, 14300, 11700, 15600]

print("Количество:", len(daily_revenue))
print("Сумма:", sum(daily_revenue))
print("Среднее:", sum(daily_revenue) / len(daily_revenue))
print("Минимум:", min(daily_revenue))
print("Максимум:", max(daily_revenue))
print("По возрастанию:", sorted(daily_revenue))

Количество: 5
Сумма: 63900
Среднее: 12780.0
Минимум: 9800
Максимум: 15600
По возрастанию: [9800, 11700, 12500, 14300, 15600]


## 12. Словари

Словарь хранит пары `ключ: значение`.

Словарь удобно использовать как карточку одного заказа.

In [35]:
order = {
    "order_id": "ORD-001",
    "city": "Москва",
    "amount": 12500,
    "is_paid": True
}

print(order)
print(order["city"])
print(order["amount"])

{'order_id': 'ORD-001', 'city': 'Москва', 'amount': 12500, 'is_paid': True}
Москва
12500


Значение можно изменить или добавить новый ключ.

In [36]:
order["amount"] = 13000
order["category"] = "Крупный"

print(order)

{'order_id': 'ORD-001', 'city': 'Москва', 'amount': 13000, 'is_paid': True, 'category': 'Крупный'}


### Список словарей

Несколько словарей можно объединить в список. Это уже похоже на небольшую таблицу.

In [37]:
orders = [
    {"order_id": "ORD-001", "city": "Москва", "amount": 12500},
    {"order_id": "ORD-002", "city": "Казань", "amount": 4800},
    {"order_id": "ORD-003", "city": "Самара", "amount": 9200}
]

print(orders)

[{'order_id': 'ORD-001', 'city': 'Москва', 'amount': 12500}, {'order_id': 'ORD-002', 'city': 'Казань', 'amount': 4800}, {'order_id': 'ORD-003', 'city': 'Самара', 'amount': 9200}]


## 13. Цикл `for`

Цикл `for` выполняет действие для каждого элемента последовательности.

In [38]:
cities = ["Москва", "Казань", "Самара"]

for city in cities:
    print(city)

Москва
Казань
Самара


### Пример 8. Суммирование в цикле

In [39]:
order_amounts = [3500, 9200, 12800, 5700, 14600]

total_revenue = 0

for amount in order_amounts:
    total_revenue = total_revenue + amount

print("Общая выручка:", total_revenue)

Общая выручка: 45800


### Пример 9. Подсчёт по условию

In [40]:
order_amounts = [2500, 7800, 12500, 4300, 15800]

large_orders_count = 0

for amount in order_amounts:
    if amount > 10000:
        large_orders_count = large_orders_count + 1

print("Количество крупных заказов:", large_orders_count)

Количество крупных заказов: 2


### Пример 10. Работа со списком словарей

In [41]:
orders = [
    {"order_id": "ORD-001", "city": "Москва", "amount": 12500},
    {"order_id": "ORD-002", "city": "Казань", "amount": 4800},
    {"order_id": "ORD-003", "city": "Самара", "amount": 9200}
]

for order in orders:
    print(f'{order["order_id"]}: {order["city"]}, {order["amount"]} руб.')

ORD-001: Москва, 12500 руб.
ORD-002: Казань, 4800 руб.
ORD-003: Самара, 9200 руб.


### Практика 4. Найдите заказы из Москвы

Используйте список `orders` из предыдущего примера.  
Выведите только идентификаторы заказов из Москвы.

In [42]:
# Сначала попробуйте решить самостоятельно

### Решение

In [43]:
orders = [
    {"order_id": "ORD-001", "city": "Москва", "amount": 12500},
    {"order_id": "ORD-002", "city": "Казань", "amount": 4800},
    {"order_id": "ORD-003", "city": "Москва", "amount": 9200}
]

for order in orders:
    if order["city"] == "Москва":
        print(order["order_id"])

ORD-001
ORD-003


## 14. Цикл `while`

Цикл `while` работает, пока условие истинно.

Внутри цикла обязательно меняйте переменную, которая участвует в условии.

In [44]:
counter = 1

while counter <= 5:
    print(counter)
    counter = counter + 1

1
2
3
4
5


### Пример 11. Накопление до порога

In [45]:
daily_revenue = [12000, 15000, 18000, 9000]
target = 40000

total = 0
day = 0

while total < target and day < len(daily_revenue):
    total = total + daily_revenue[day]
    day = day + 1

print("Накопленная выручка:", total)
print("Понадобилось дней:", day)

Накопленная выручка: 45000
Понадобилось дней: 3


## 15. Короткое знакомство с функциями

Функция объединяет повторяющиеся действия в один именованный блок.

In [46]:
def calculate_revenue(unit_price, quantity):
    revenue = unit_price * quantity
    return revenue

result = calculate_revenue(2500, 4)

print(result)

10000


### Пример 12. Функция расчёта суммы со скидкой

In [47]:
def calculate_final_amount(unit_price, quantity, discount_rate=0):
    amount = unit_price * quantity
    final_amount = amount * (1 - discount_rate)
    return final_amount

print(calculate_final_amount(2500, 3))
print(calculate_final_amount(2500, 3, 0.10))

7500
6750.0


## 16. Итоговая мини-практика

Даны заказы интернет-магазина.

Нужно:

1. посчитать общую выручку;
2. найти средний чек;
3. посчитать оплаченные заказы;
4. посчитать крупные заказы больше 10 000 рублей;
5. определить, выполнен ли план 40 000 рублей.

In [48]:
orders = [
    {"order_id": "ORD-001", "amount": 3500, "is_paid": True},
    {"order_id": "ORD-002", "amount": 9200, "is_paid": True},
    {"order_id": "ORD-003", "amount": 12800, "is_paid": False},
    {"order_id": "ORD-004", "amount": 5700, "is_paid": True},
    {"order_id": "ORD-005", "amount": 14600, "is_paid": True}
]

plan = 40000

total_revenue = 0
paid_orders_count = 0
large_orders_count = 0

for order in orders:
    total_revenue = total_revenue + order["amount"]

    if order["is_paid"]:
        paid_orders_count = paid_orders_count + 1

    if order["amount"] > 10000:
        large_orders_count = large_orders_count + 1

average_check = total_revenue / len(orders)

if total_revenue >= plan:
    plan_status = "План выполнен"
else:
    plan_status = "План не выполнен"

print(f"Количество заказов: {len(orders)}")
print(f"Оплаченных заказов: {paid_orders_count}")
print(f"Общая выручка: {total_revenue} руб.")
print(f"Средний чек: {average_check:.2f} руб.")
print(f"Крупных заказов: {large_orders_count}")
print(plan_status)

Количество заказов: 5
Оплаченных заказов: 4
Общая выручка: 45800 руб.
Средний чек: 9160.00 руб.
Крупных заказов: 2
План выполнен


## Контрольные вопросы

1. Чем `int` отличается от `float`?
2. Почему `"10" + "20"` даёт `"1020"`?
3. Чем `=` отличается от `==`?
4. Что делает `and`?
5. С какого индекса начинается список?
6. Для чего используется словарь?
7. Чем `for` отличается от `while`?
8. Почему в условии и цикле важны отступы?

## Самопроверка

После занятия вы должны уметь:

- создавать переменные;
- определять и преобразовывать типы;
- выполнять расчёты;
- форматировать текст;
- использовать сравнения и логические операторы;
- писать условия;
- работать со списками и словарями;
- применять `for` и простой `while`;
- написать простую функцию.